src/verifytime.c

```diff
@@ -0,0 +1,96 @@
+/*
+ *   Software Updater - client side
+ *
+ *      Copyright © 2012-2016 Intel Corporation.
+ *
+ *   This program is free software: you can redistribute it and/or modify
+ *   it under the terms of the GNU General Public License as published by
+ *   the Free Software Foundation, version 2 or later of the License.
+ *
+ *   This program is distributed in the hope that it will be useful,
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *   GNU General Public License for more details.
+ *
+ *   You should have received a copy of the GNU General Public License
+ *   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ *   Authors:
+ *         Tudor Marcu <tudor.marcu@intel.com>
+ *
+ */
+#include <stdbool.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <sys/stat.h>
+#include <time.h>
+
+#define DAY_SECONDS 86400
+
+unsigned long int get_versionstamp(void)
+{
+	struct stat statt;
+	FILE *fp = NULL;
+	char data[11];
+	const char *filename = "/usr/share/clear/versionstamp";
+
+	if (stat(filename, &statt) == -1) {
+		printf("%s does not exist!\n", filename);
+		return 0;
+	}
+
+	fp = fopen(filename, "r");
+	if (fp == NULL) {
+		printf("Failed to open %s\n", filename);
+		return 0;
+	}
+
+	if (fgets(data, 11, fp) == NULL) {
+		printf("Failed to read %s\n", filename);
+		fclose(fp);
+		return 0;
+	}
+
+	fclose(fp);
+	return strtoul(data, NULL, 10);
```

Comment:
Since a 0 return value indicates an error condition, if the versionstamp value is 0, we should handle that case.